File: src/models/new-discussion-recipient.ts
import type { EntityKind } from "./entity-kind";
import type { NewDiscussionRecipientFunction } from "./new-discussion-recipient-function";
import type { NewDiscussionRecipientSubject } from "./new-discussion-recipient-subject";

export type NewDiscussionRecipient = Readonly<{
  name: string;
  kind: typeof EntityKind.Teacher | typeof EntityKind.Student | typeof EntityKind.Personal;
  id: string;
  isPrincipal: boolean;
  subjects: Array<NewDiscussionRecipientSubject>;
  function?: NewDiscussionRecipientFunction;
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/NewDiscussionRecipient.kt
package ink.literate.pawnote.models

data class NewDiscussionRecipient(
    val name: String,
    val kind: DiscussionRecipientKind,
    val id: String,
    val isPrincipal: Boolean,
    val subjects: List<NewDiscussionRecipientSubject>,
    val function: NewDiscussionRecipientFunction? = null
)